opentracing/span.py

```diff
@@ -73,6 +73,7 @@ class Span(object):
     def __init__(self, tracer, context):
         self._tracer = tracer
         self._context = context
+        self._deactivate_on_finish = False
```

Comment:
Here the `_deactivate_on_finish` is evaluated to decide if the `source.deactivate(span)` must be called or not when calling `span.finish()`. 
In the current implementation that field is updated in the `ActiveSpanSource` implementation when a `Span` is activated (ref: [some implementations][1]).

If we keep it, we should probably set it to True at least in the noop implementation, so that it's clear why the attribute is here.

As alternative, we may get rid of this attribute and say "all spans are deactivated in any case, and the `source.deactivate()` must handle that".

[1]: https://github.com/palazzem/ot-examples/blob/master/ext/active_span_source.py#L28